### PR TITLE
Add mfa, possible fix for #140

### PR DESCRIPTION
**Pull request by xbenjii**
Apologies as I'm not well versed in C#, also needs the new tests adding but this should give us a start.

Let me know if you need anything else.

### CODE REVIEW

File: MegaApiClient/MegaApiClient.cs [hunk 1]
```diff
@@ -183,7 +209,14 @@ public LogonSessionToken Login(AuthInfos authInfos)
       this.authenticatedLogin = true;
 
       // Request Mega Api
-      LoginRequest request = new LoginRequest(authInfos.Email, authInfos.Hash);
+      LoginRequest request;
+      if (!string.IsNullOrEmpty(authInfos.MFAKey))
+      {
+        request = new LoginRequest(authInfos.Email, authInfos.Hash, authInfos.MFAKey);
+      } else
```
Comment (gpailler) [inline MegaApiClient/MegaApiClient.cs:216]: Could you move the `else` on a dedicated line?

File: MegaApiClient/Serialization/Login.cs [hunk 2]
```diff
@@ -11,11 +11,22 @@ public LoginRequest(string userHandle, string passwordHash)
       this.PasswordHash = passwordHash;
     }
 
+    public LoginRequest(string userHandle, string passwordHash, string mfaKey)
+  : base("us")
```
Comment (gpailler) [inline MegaApiClient/Serialization/Login.cs:15]: Could you indent like the other constructor?

**Comment by gpailler** [comment]
Hi @xbenjii . Thanks a lot for the PR. I tested quickly and it works perfectly. I think adding unit tests for MFA will be complicated so I'm ok to merge the PR without any additional test.
Could just handle the styling comments?

Thanks!

**Comment by xbenjii** [comment]
@gpailler Thanks! I was a bit concerned with the failing unit tests but then realized they require a bit of setup. Not sure about the flaky ones in CI too, as they sometimes work and other times don't.

**Comment by gpailler** [comment]
Yes, I will check the tests in the CI. They are sometimes failing because of Mega API throttling.

**Comment by gpailler** [comment]
Will merge to master and will fix the UTs

**Pull request merged**